include/cpp-statsd-client/UDPSender.hpp

```diff
@@ -165,12 +213,16 @@ inline UDPSender::~UDPSender() {
 
     // If we're running a background thread tell it to stop
     if (m_batchingThread.joinable()) {
-        m_mustExit.store(true, std::memory_order_acq_rel);
+        m_mustExit.store(true, std::memory_order_release);
```

Comment:
i had to change these acquire on load and release on store otherwise windows would give back "invalid memory order"

Reply:
Hmm ok. Do you have more info about why this segfault occurs?

Reply:
seems to occur when the `load` gets to the `release` part. here are the messages i see in visual studio in order:

![image](https://user-images.githubusercontent.com/697548/125289598-697f2980-e2ed-11eb-9009-f4162a4e6b5a.png)

![image](https://user-images.githubusercontent.com/697548/125289615-70a63780-e2ed-11eb-80ab-8ad44434aefd.png)

![image](https://user-images.githubusercontent.com/697548/125289637-7734af00-e2ed-11eb-86d1-27f033d08803.png)

Reply:
so it hits an assertion in debug mode which would give me the intuition that we are doing something specifically wrong. then i click ignore to see what will happen next and it throws an exception so i look at the call stack and we can see it was in the lambda defined in the constructor. anyway when i first saw this i went and read the docs and found this bit:

https://en.cppreference.com/w/cpp/atomic/memory_order

> **Release-Acquire ordering**
> If an atomic store in thread A is tagged memory_order_release and an atomic load in thread B from the same variable is tagged memory_order_acquire, all memory writes (non-atomic and relaxed atomic) that happened-before the atomic store from the point of view of thread A, become visible side-effects in thread B. That is, once the atomic load is completed, thread B is guaranteed to see everything thread A wrote to memory.

This sounded just like what we were looking for so i changed the implementation to match that suggestion.

Reply:
ah here is the answer in the docs for load and store:

https://en.cppreference.com/w/cpp/atomic/atomic/load
> order must be one of std::memory_order_relaxed, std::memory_order_consume, std::memory_order_acquire or std::memory_order_seq_cst. Otherwise the behavior is undefined.

https://en.cppreference.com/w/cpp/atomic/atomic/store
> order must be one of std::memory_order_relaxed, std::memory_order_release or std::memory_order_seq_cst. Otherwise the behavior is undefined.